src/tests/ImportNameRuleTests.ts

```diff
@@ -271,4 +241,119 @@ describe('importNameRule', (): void => {
+                        name: Utils.absolutePath('file.ts'),
+                        ruleName: 'import-name',
+                        startPosition: { character: 21, line: 2 },
+                        fix: {
+                            innerStart: 28,
+                            innerLength: 12,
+                            innerText: 'userSettingsDetailView'
+                        }
+                    }
+                ]);
+            });
+        });
+
+        describe('pascalCase', (): void => {
+            it('should pass on matching names of ES6 import', (): void => {
+                const script: string = `
+                    import App from 'App';
+                    import App from 'x/y/z/App';
+                    import GraphqlTag from 'graphql-tag'
+                `;
+
+                const options = [
+                    true,
+                    {},
+                    {},
+                    {
+                        case: 'PascalCase'
+                    }
+                ];
+
+                TestHelper.assertViolationsWithOptions(ruleName, options, script, []);
+            });
+
+            it('should fail import with punctuation and underscore', (): void => {
+                const script: string = `
+                    import UserSettings from "./user-settings.detail_view";
+                `;
+
+                const options = [
+                    true,
+                    {},
+                    {},
+                    {
+                        case: 'PascalCase'
+                    }
+                ];
+
+                TestHelper.assertViolationsWithOptions(ruleName, options, script, [
+                    {
+                        failure: "Misnamed import. Import should be named 'UserSettingsDetailView' but found 'UserSettings'",
+                        name: Utils.absolutePath('file.ts'),
+                        ruleName: 'import-name',
+                        startPosition: { character: 21, line: 2 },
+                        fix: {
+                            innerStart: 28,
+                            innerLength: 12,
+                            innerText: 'UserSettingsDetailView'
+                        }
+                    }
+                ]);
+            });
+        });
+
+        describe('any-case', () => {
+            it('should pass on PascalCase and camelCase', () => {
+                const script: string = `
+                    import App from 'app';
+                    import app from 'app';
+                    import App from 'x/y/z/App';
+                    import GraphqlTag from 'x/y/z/graphql-tag';
+                    import graphqlTag from 'x/y/z/graphql-tag';
+                `;
+                const options = [true, {}, {}, { case: 'any-case' }];
+                TestHelper.assertViolationsWithOptions(ruleName, options, script, []);
```

Comment:
I think it will be nice to have example with error for `any-case` options.

P.S. Error case would also show problem with fix in this rule.